frontend/components/MainContent/MainContent.tsx

```diff
@@ -45,17 +50,24 @@ const MainContent = ({
   const showAppleABMBanner =
     isAppleBmTermsExpired && isPremiumTier && !isSandboxMode;
 
+  // ABM banner takes precedence
+  const showLicenseExpirationBanner =
+    isLicenseExpired && isPremiumTier && !isAppleBmTermsExpired;
+
   return (
     <div className={classes}>
-      {showAppleABMBanner && <AppleBMTermsMessage />}
-      <SandboxGate
-        fallbackComponent={() => (
-          <SandboxExpiryMessage
-            expiry={sandboxExpiryTime}
-            noSandboxHosts={noSandboxHosts}
-          />
-        )}
-      />
+      <div className={`${baseClass}--animation-disabled`}>
```

Comment:
Would [`React.memo`](https://react.dev/reference/react/memo) izing the components prevent those undesired rerenders/animations?